src/main/java/seedu/address/model/property/PriceAndOneTagsPredicate.java

```diff
@@ -27,8 +27,12 @@ public PriceAndOneTagsPredicate(Price price, Set<Tag> tags) {
 
     @Override
     public boolean test(Property property) {
-        return tags.stream().anyMatch(tag -> property.getTags().contains(tag))
-                && property.getPrice().isInRangePrice(price);
+        if (tags.size() == 0) {
+            return property.getPrice().isInRangePrice(price);
+        } else {
+            return tags.stream().anyMatch(tag -> property.getTags().contains(tag))
+                    && property.getPrice().isInRangePrice(price);
+        }
```

Comment:
I think you can simplify it to be like this:

```suggestion
        return (tags.size() == 0 || tags.stream().anyMatch(tag -> property.getTags().contains(tag)))
            && property.getPrice().isInRangePrice(price);
```